### PR TITLE
4.x translate collections

### PR DESCRIPTION
**Pull request by tavaresgerson**

### CODE REVIEW

File: pt/core-libraries/caching.rst [hunk 1]
```diff
@@ -1,17 +1,614 @@
...
+* ``Null`` O mecanismo nulo não armazena nada e falha em todas as operações de leitura. 
+
+Independentemente do CacheEngine que você escolher, seu aplicativo interage com
+:php:class:`Cake\\Cache\\Cache`.
+
 .. _cache-configuration:
 
-Configuring Cache Class
-=======================
+Configuring Cache Engines
```
Comment (othercorey) [inline pt/core-libraries/caching.rst:44]: This is all english. We don't want to have english in translation pages because it is never updated.

**Comment by markstory** [comment]
@tavaresgerson  Do you want to give us a ping when this is ready to go?

**Comment by tavaresgerson** [comment]
> @tavaresgerson Do you want to give us a ping when this is ready to go?

Yes :smile:

**Comment by tavaresgerson** [comment]
Done translate! :sunglasses:

File: pt/core-libraries/caching.rst [hunk 2]
```diff
@@ -1,17 +1,611 @@
...
+
+    // Namespace completo
+    Cache::setConfig('long', ['className' => 'Cake\Cache\Engine\FileEngine']);
+
+    // Um objeto implementando CacheEngineInterface
+    Cache::setConfig('long', ['className' => $myCache]);
+
+.. note::
+
+    Ao usar o FileEngine, pode ser necessário usar a opção ``mask`` para
```
Comment (othercorey) [inline pt/core-libraries/caching.rst:130]:
```suggestion
    Ao usar o ``FileEngine``, pode ser necessário usar a opção ``mask`` para 
```

File: pt/core-libraries/caching.rst [hunk 3]
```diff
@@ -1,17 +1,611 @@
...
+   configuração. Útil quando você precisa excluir um subconjunto de dados de um cache.
+* ``prefix`` Anexado a todas as entradas. É bom para quando você precisar compartilhar 
+   um espaço de chave com outra configuração de cache ou outro aplicativo.
+* ``probability`` Probabilidade de atingir uma limpeza de cache. Definir como 0 o
+   ``Cache::gc()`` será desativado e não funcionará automaticamente.
+
+Opções do FileEngine
+--------------------
+
+O FileEngine usa as seguintes opções específicas do mecanismo:
```
Comment (othercorey) [inline pt/core-libraries/caching.rst:150]:
```suggestion
O ``FileEngine`` usa as seguintes opções específicas do mecanismo:
```

File: pt/core-libraries/caching.rst [hunk 4]
```diff
@@ -1,17 +1,611 @@
...
+
+    // Decrementa
+    Cache::decrement('initial_count');
+
+    // Ou
+    Cache::increment('initial_count');
+
+.. note::
+
+    Incrementar e decrementar não funcionam com o FileEngine.
```
Comment (othercorey) [inline pt/core-libraries/caching.rst:443]:
```suggestion
    Incrementar e decrementar não funcionam com o ``FileEngine``. 
```

File: pt/core-libraries/collections.rst [hunk 5]
```diff
@@ -1195,5 +1181,5 @@ método ``compile()``::
     }
 
 .. meta::
-    :title lang=pt-br: Coleções
-    :keywords lang=pt-br: coleções, cakephp, append, sort, compile, contains, countBy, each, every, extract, filter, first, firstMatch, groupBy, indexBy, jsonSerialize, map, match, max, min, reduce, reject, sample, shuffle, some, random, sortBy, take, toArray, insert, sumOf, stopWhen, unfold, through
+    :title lang=pt: Coleções
```
Comment (othercorey) [inline pt/core-libraries/collections.rst:1184]: Why change `pt-br` to `pt`?

File: pt/core-libraries/caching.rst [hunk 6]
```diff
@@ -1,17 +1,611 @@
...
+    :return: Booleano ``true`` para sucesso.
+
+    Incremente um número abaixo da chave e retorna valor incrementado
 
-    Você pode consultar a versão em inglês deste tópico através do seletor de
-    idiomas localizado ao lado direito do campo de buscas da documentação.
\ No newline at end of file
+.. meta::
+    :title lang=pt: Cache
+    :keywords lang=pt-br: uniforme api,cache engine,sistema de cache,operacoes atomicas,php class,armazenamento em disco,metodos estaicos,extensao php,consistencia,recursos similares,apcu,apc,memcache,consultas,cakephp,elementos,servidores,memoria
```
Comment (othercorey) [inline pt/core-libraries/caching.rst:611]: `pt` or `pt-br` ?

File: pt/core-libraries/collections.rst (hunk 5 above)
Comment (tavaresgerson) [reply pt/core-libraries/collections.rst:1184]: Excuse-me

File: pt/core-libraries/caching.rst (hunk 6 above)
Comment (tavaresgerson) [reply pt/core-libraries/caching.rst:611]: `pt`

**Pull request merged**